### PR TITLE
Remove deprecated slurm options `MEMORY` and `MEMORY_PER_CPU`

### LINKED ISSUE #9465: test_res_initialization.py::test_everest_to_ert_queue_config fails when a site config specifies `REALIZATION_MEMORY`
This test will fail if there is a site_config which sets `REALIZATION_MEMORY` and the user has specified the `SLURM` option: `MEMORY`. Fixing this means making sure that the user specified `MEMORY` will overwrite the site config specified `REALIZATION_MEMORY`. If the user has specified both and none come from the site config the current error is ok.

To reproduce with the poly example, add:

```bash
QUEUE_OPTION SLURM MEMORY
``` 

and modify the site config by adding:

```python
"REALIZATION_MEMORY 100"
```

Image

### PR DESCRIPTION
**Pull request by jonathan-eq**
**Issue**
Resolves #9465

**Approach**
✂️ 

(Screenshot of new behavior in GUI if applicable)

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/ert/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/config/queue_config.py [hunk 1]
```diff
@@ -350,19 +338,6 @@ def from_dict(cls, config_dict: ConfigDict) -> QueueConfig:
                     "MEMORY_PER_JOB",
```
[the remark this answers is not in the document]
Comment (jonathan-eq) [reply src/ert/config/queue_config.py:338]: i am only doing the slurm options in this commit. I have a different PR on the torque options, and will create a new issue for the lsf options 👌

File: src/ert/scheduler/slurm_driver.py [hunk 2]
```diff
@@ -148,8 +135,6 @@ def _submit_cmd(
             # In slurm, --mem==0 requests all memory on a node. In Ert,
             # zero realization memory means no intended memory allocation.
             sbatch_with_args.append(f"--mem={self._realization_memory // 1024**2}M")
```
Comment (xjules) [inline src/ert/scheduler/slurm_driver.py:137]: How is the compatible with all the:
```python
torque_memory_usage_format: QueueMemoryStringFormat = QueueMemoryStringFormat(
    suffixes=["kb", "mb", "gb", "KB", "MB", "GB"]
)
``` 
as this one assumes MB?

**Review by xjules — CHANGES_REQUESTED** [review]
Looks very good! Had a question on the different memory formats.

File: src/ert/scheduler/slurm_driver.py (hunk 2 above)
Comment (xjules) [reply src/ert/scheduler/slurm_driver.py:137]: Clarified via discussion.

**Review by xjules — APPROVED** [review]
Let's :scissors: :rocket:

**Pull request merged**